### PR TITLE
OPS-8292 add to stub basic flows from accreditation

### PR DESCRIPTION
**Pull request by gitwojciech**

### CODE REVIEW

File: app/actors/NoSurchargeMessageFlowUserActor.scala [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+
+object NoSurchargeMessageFlowUserActor {
+  def props(spcFlow:SpcFlow):Props = Props(new NoSurchargeMessageFlowUserActor(spcFlow))
+}
+
+
+class NoSurchargeMessageFlowUserActor(spcFlow:SpcFlow) extends Actor {
+  import SpcParentActor._
+
+  var schedule:Cancellable = startCountDown()
```
Comment (ed1286) [inline app/actors/NoSurchargeMessageFlowUserActor.scala:35]: is using var ok?

File: app/models/AmountInPence.scala [hunk 2]
```diff
@@ -42,10 +42,10 @@ object AmountInPence {
   def apply(str: String): AmountInPence = {
     str match {
       case s if s.isEmpty => AmountInPence(0)
-      case s              => AmountInPence(BigDecimal(s).longValue() * 100)
```
Comment (ed1286) [inline app/models/AmountInPence.scala:45]: if this means that AmountInPence is different to how it is on most other services then I'd rename the class to make that obvious.

File: app/actors/NoSurchargeMessageFlowUserActor.scala (hunk 1 above)
Comment (gitwojciech) [reply app/actors/NoSurchargeMessageFlowUserActor.scala:35]: yes using var. There are other options but will make code less readable

File: app/models/PaymentCard.scala [hunk 3]
```diff
@@ -18,8 +18,24 @@ package models
...
+                        endDate: String,
+                        startDate:String,
+                        pan:String,
+                        cardSchema: CardSchema,
+                        seqNum:String,
+                        availableSpend: Option[AmountInPence]) {
+
+  def receiptPan: String = pan.take(6) + pan.drop(6).replaceAll(".(?=.{4})", "*")
+  def receiptPanMasked: String = "*** Data Removed for Security ***"
+  def receiptStart: String = new SimpleDateFormat("MM/yy").format( new SimpleDateFormat("yyyy-MM-dd").parse(startDate).getTime)
```
Comment (ed1286) [inline app/models/PaymentCard.scala:34]: out of interest what exactly is happening here?

File: app/actors/NoSurchargeMessageFlowUserActor.scala (hunk 1 above)
Comment (ed1286) [reply app/actors/NoSurchargeMessageFlowUserActor.scala:35]: as it's a stub it's ok I guess

File: app/models/AmountInPence.scala (hunk 2 above)
Comment (gitwojciech) [reply app/models/AmountInPence.scala:45]: Stub and  F2F keep money in pence. That should be in line with other systems.

**Review by ed1286 — APPROVED** [review]

**Pull request merged**